### DISCUSSION TITLE
Is there a way to preprocess URL query result before actual Json parsing?

### DISCUSSION THREAD

**Discussion by imReker** [grafana/grafana-infinity-datasource#331] [General]
Some API of Google will add a prefix `)]}'` to the JSON result, for example:
https://transparencyreport.google.com/transparencyreport/api/v3/traffic/fraction?start=1652745600000&end=1655423999999&region=VI&product=19

Is there a way to remove the prefix before Infinity parse the JSON?

**Comment by yesoreyeram** [comment]
I haven't seen such a weird format, especially from google. What format is this?

BTW, Sorry currently I couldn't find any possible ways. But potentially possible in future versions with UQL.

**Reply by imReker** [reply]
It's JSON, but with a weird prefix. I reviewed the JS source, it just ignored the `)]}'` prefix.
```
if (a.startsWith(")]}'\n") && (a = a.substring(5))){
    return JSON.parse(a)[0];
}
```

**Reply by yosiasz** [reply]
it is JSON but an invalid one
